### PR TITLE
* added django-1.9 support

### PR DESCRIPTION
**Pull request by randomknowledge**
* dropped support for django-cms-3.1.x, django-1.6 and python-2.6

### CODE REVIEW

File: barbeque/tests/test_commands.py [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import pytest
-from django.utils.datastructures import SortedDict
+try:
+    from django.utils.datastructures import SortedDict
```
Comment (stephrdev) [inline barbeque/tests/test_commands.py]: We dropped Python 2.6 support. 2.7 supports OrderedDict - why don't we use OrderedDict all the time?

File: barbeque/tests/views/test_mixins.py [hunk 2]
```diff
@@ -59,7 +59,7 @@ def test_logged_in(self, rf, settings):
         request.user = get_user_model()(username='testuser')
 
         response = protected_view(request)
-        assert response.status_code == 301
+        assert response.status_code == 302 or response.status_code == 301
```
Comment (stephrdev) [inline barbeque/tests/views/test_mixins.py]: I think we should have only one valid status_code. Why do they differ?

File: setup.py [hunk 3]
```diff
@@ -69,7 +70,7 @@ def run_tests(self):
     packages=find_packages(exclude=['barbeque.tests']),
     test_suite='.',
     tests_require=tests_require,
-    install_requires=['Django>=1.6,<1.9'],
+    install_requires=['Django>=1.6,<1.10'],
```
Comment (stephrdev) [inline setup.py]: I think we should update the install_requires to as we dropped Django 1.6 support.

File: barbeque/tests/views/test_mixins.py (hunk 2 above)
Comment (randomknowledge) [inline barbeque/tests/views/test_mixins.py]: The default value of the permanent attribute in RedirectView changed from True to False in Django 1.9.

Comment (stephrdev) [inline barbeque/tests/views/test_mixins.py]: Lets fix the DummyView then py explicitly passing the permanent flag.

**Pull request merged**